crates/datasources/src/sqlite/convert.rs

```diff
@@ -0,0 +1,217 @@
+                                            from: Value::Integer(*i),
+                                            to: DataType::Float64,
+                                        })
+                                    }
+                                }
+                                Value::Real(r) => Ok(Some(*r)),
+                                Value::Text(t) => {
+                                    if t.is_empty() || t.eq_ignore_ascii_case("null") {
+                                        Ok(None)
+                                    } else {
+                                        t.parse::<f64>().map(Some).map_err(|_| {
+                                            SqliteError::InvalidConversion {
+                                                from: Value::Text(t.clone()),
+                                                to: DataType::Float64,
+                                            }
+                                        })
+                                    }
+                                }
+                                v => Err(SqliteError::InvalidConversion {
+                                    from: v.clone(),
+                                    to: DataType::Float64,
+                                }),
+                            }
+                        })
+                        .collect::<Result<Float64Array>>()?,
+                ),
+                DataType::Utf8 => {
+                    let mut builder = StringBuilder::with_capacity(data.len(), 10 * data.len());
+                    let mut buf = String::new();
+                    for row in data.iter() {
+                        match row
+                            .get(col_idx)
+                            .ok_or_else(|| SqliteError::MissingDataForColumn(col_idx))?
+                        {
+                            Value::Null => builder.append_null(),
+                            Value::Integer(i) => {
+                                buf.clear();
+                                encode::encode_int(&mut buf, *i)?;
+                                builder.append_value(&buf);
+                            }
+                            Value::Real(r) => {
+                                buf.clear();
+                                encode::encode_float(&mut buf, *r)?;
+                                builder.append_value(&buf);
+                            }
+                            Value::Text(t) => builder.append_value(t),
+                            Value::Blob(b) => {
+                                buf.clear();
+                                encode::encode_binary(&mut buf, b)?;
+                                builder.append_value(&buf);
+                            }
+                        }
+                    }
+                    Arc::new(builder.finish())
+                }
+                DataType::Binary => {
+                    let mut builder = BinaryBuilder::with_capacity(data.len(), 10 * data.len());
```

Comment:
why 10x?

Reply:
Just an estimate we've used. Assuming average length of 1 blob to be 10 bytes.